packages/astro/src/cli/check/index.ts

```diff
@@ -17,9 +18,12 @@ interface Result {
 	hints: number;
 }
 
-export async function check(settings: AstroSettings) {
+export async function check(settings: AstroSettings, { logging }: { logging: LogOptions }) {
 	console.log(bold('astro check'));
 
+	const { sync } = await import('../sync/index.js');
+	await sync(settings, { logging, fs });
```

Comment:
```suggestion
	const syncRet = await sync(settings, { logging, fs });
	// early exit on sync failure
	if (syncRet === 1) return syncRet;
```
Super pedantic, but check for an early `process.exit` if necessary!